### DISCUSSION TITLE
java 21

### DISCUSSION THREAD

**Discussion by MasterOfTouch** [bitfireAT/ical4android#193] [General]
Is there a reason why lib was udpated to use java 21?
https://github.com/bitfireAT/ical4android/commit/eb50f30122721c2314d0249b7b873b6cc0da95e0
Just curious what is the reason to push a such high numbers so early.
This library is not easy to use in non-hipster projects.

**Comment by rfc2822** [comment]
> Is there a reason why lib was udpated to use java 21?

Current [Android Studio comes with Java 21](https://android-developers.googleblog.com/2024/04/android-studio-jellyfish-is-stable.html?debug=true) and encourages its use.

>  This library is not easy to use in non-hipster projects.

I don't know what "hipster projects" are, but are you not using Android Studio? It would be helpful to know your specific use-case.

**Reply by MasterOfTouch** [reply]
Using latest stable Android studio not forcing you to use embeded jbr.
But you don't need to downlaod different jdk either.
The default version of JDK for Android Gradle plugin 8.7 is 17.

**Reply by MasterOfTouch** [reply]
Since it's a library, the projects that not udpated to java 21 can't use it.
Some release tags could be added at least, so users can aim specific version that suit.

**Reply by MasterOfTouch** [reply]
By hipster project I mean a project that keep everything updated as much as possible with tons of libs that released yesterday, while big mature projects are in no rush to update everything as quickly as possible, they fail to keep up with the times and adopt all the available technologies.
Sorry for confusing.

**Reply by ArnyminerZ** [reply]
>  Some release tags could be added at least, so users can aim specific version that suit.

Hi, we use [Jitpack](https://jitpack.io/#bitfireAT/ical4android) for releases. From there, you can select the commit you want to use.
